packages/x-charts/src/ChartContainer/ResizableContainer.tsx

```diff
@@ -34,15 +34,15 @@ export const ResizableContainerRoot = styled('div', {
  * @ignore - do not document.
  */
 function ResizableContainer(props: { children: React.ReactNode }) {
-  const { inHeight, inWidth, hasIntrinsicSize, containerRef } = useSize();
+  const { inHeight, inWidth, containerRef } = useSize();
 
   return (
     <ResizableContainerRoot
       {...props}
       ownerState={{ width: inWidth, height: inHeight }}
       ref={containerRef}
     >
-      {hasIntrinsicSize && props.children}
+      {props.children}
```

Comment:
That let me think that the initial idea of this early return was to skip providers on first render.

If you have no width/height, you will have default width/height used to compute scales, coordinate, ... And when it renders, the `width`/`height` get their real size and everything run a second time. The idea was to skip the useless step.

Reply:
Providers are now on top of this component, since the datasource is the `<ChartDataProvider>` now. So I moved the check to the `ChartsSurface`. This means we can render the `<svg/>` TAG on the first render, but everything else is only rendered if we have an intrinsic size